### Ruby's raise vs throw
* raise, fail, rescue, and ensure handle errors, also known as *exceptions*
* throw and catch are control flow; they emulate a "go to": Ruby's throw/catch lets you break execution and climb up the stack looking for a catch (like raise/rescue does), but isn't really meant for error conditions. It should be used rarely, and is there just for when the "walk up the stack until you find a corresponding catch" behaviour makes sense for an algorithm you're writing but it wouldn't make sense to think of the throw as corresponding to an error condition.
(you can use it if for example you wanna break 16 anidated structures or the snippet you wanna execute is far far away, maybe in another method)